File: Data Structures & Algorithms/remove-node-from-end-of-linked-list/submission-1.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */

class Solution {
public:
    ListNode* removeNthFromEnd(ListNode* head, int n) {
       
        ListNode* temp = head;
        ListNode* prev = NULL;
        int cnt  = 0;
        while(temp!=NULL){
            cnt++;
            temp=temp->next;
        }
        if(n==1 && cnt==1) return NULL;
        int n1 = cnt - n;
        if(n1==0){
        ListNode* newHead = head->next;
        delete head;
        return newHead;
        }
        temp = head;
        int j = 0;
        while(temp!=NULL && j<=cnt){
            if(j == n1){
              prev->next = prev->next->next;  
             }
            j++;
            prev = temp;
            temp = temp->next;
        }
        return head;
    }
};

File: Data Structures & Algorithms/reorder-linked-list/submission-1.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next

class Solution:
    def reorderList(self, head: Optional[ListNode]) -> None:
        if not head: return head
        temp = head
        sta = []
        while temp!=None:
            sta.append(temp)
            temp = temp.next
        total = len(sta)//2
        limit  = len(sta) - total
        temp = head
        while temp!=None and len(sta)>limit:
            front = temp.next
            add = sta.pop()
            temp.next = add
            add.next = front
            temp = front
        if temp:
            temp.next = None
        